activerecord/lib/arel/visitors/sqlite.rb

```diff
@@ -58,7 +64,7 @@ def prepare_update_statement(o)
             # The SQLite3 dialect isn't flexible enough to allow anything other than a inner join
             # for the first join:
             #   UPDATE table SET .. FROM joined_table WHERE ...
-            (o.relation.right.first.is_a?(Arel::Nodes::InnerJoin))
+            (o.relation.right.all? { |join| join.is_a?(Arel::Nodes::InnerJoin) || join.right.expr.right.relation != o.relation.left })
```

Comment:
I'm not sure I understand that `||` condition, could you explain what this is trying to match for?

Reply:
When we have a join on the target (the table we are updating), the joined table has to be in the `FROM` clause for the query to be valid. This is what the first commit in this PR fixes.

By introducing this fix, we now hit the flexibility issue you previously mentioned when a left outer join is used. 
What this `||` checks, is whether there is any left joins on the target which we have to workaround and resort to making a subselect query instead.

This check is only required on left joins on the target (as we have to translate that join into a FROM and we hit the flexibility issue). Issuing a left outer join on another table is fine since we don't have to translate it for the query to be syntax valid.

Reply:
> What this `||` checks, is whether there is any left joins on the target which we have to workaround and resort to making a subselect query instead.

That is where I'm confused, this `if` condition is for case where we don't have to fallback to a a subquery. So why `||`, wasn't this meant to be `&&`? (Sorry, I'm not the best with Arel API)